test/unit/presenters/personal/personal-details-presenter.test.js

```diff
@@ -0,0 +1,98 @@
+// Test framework dependencies
+import { describe, test, expect, beforeEach, vi } from 'vitest'
+
+// Thing under test
+import { personalDetailsPresenter } from '../../../../src/presenters/personal/personal-details-presenter.js'
+
+// Mock data
+import { mappedData as originalData } from '../../../mocks/mock-personal-details.js'
+
+describe('personalDetailsPresenter', () => {
+  let yar
+  let data
+
+  beforeEach(() => {
+    vi.clearAllMocks()
+
+    // Deep clone the data to avoid mutation across tests
+    data = JSON.parse(JSON.stringify(originalData))
+
+    // Mock yar session manager
+    yar = {
+      flash: vi.fn().mockReturnValue([{ title: 'Update', text: 'Personal details updated successfully' }])
+    }
+  })
+
+  describe('when provided with personal details data', () => {
+    test('it correctly presents the data', () => {
+      const result = personalDetailsPresenter(data, yar)
+
+      expect(result).toEqual({
+        notification: { title: 'Update', text: 'Personal details updated successfully' },
+        pageTitle: 'View and update your personal details',
+        metaDescription: 'View and update your personal details.',
+        address: [
+          'THE COACH HOUSE',
+          'STOCKWELL HALL',
+          '7 HAREWOOD AVENUE',
+          'DARLINGTON',
+          'Dorset',
+          'CO9 3LS',
+          'United Kingdom'
+        ],
```

Comment:
Can we mock address-presenter here, so that this unit test does not directly depend on the outcome of address-presenter (base-presenter as it is called now)?